src/components/SpruceForm/FieldTemplates.tsx

```diff
@@ -249,6 +247,28 @@ export const ArrayFieldTemplate: React.VFC<ArrayFieldTemplateProps> = ({
   const topAlignDelete = uiSchema["ui:topAlignDelete"] ?? false;
   const useExpandableCard = uiSchema["ui:useExpandableCard"] ?? false;
   const isDisabled = disabled || readonly;
+
+  const addButtonSize = uiSchema["ui:addButtonSize"] || "small";
+  const addButtonText = uiSchema["ui:addButtonText"] || "Add";
+  // Override RJSF's default array behavior; add new elements to beginning of array unless otherwise specified.
+  const addToEnd = uiSchema["ui:addToEnd"] ?? false;
+  const handleAddClick =
+    items.length && !addToEnd ? items[0].onAddIndexClick(0) : onAddClick;
+
+  const addButton = (
```

Comment:
This should live as its own component  outside of this and ingest props otherwise it is redeclared on each render.

Reply:
Since the goal here is to have two (almost) identical buttons that render at different positions in the form, making a separate component and including it twice resulted in a lot of repetition. I thought a `useMemo` made more sense here so that any future changes to the button would be reflected in both instances.